x-pack/solutions/security/plugins/security_solution_serverless/server/product_features/index.ts

```diff
@@ -39,31 +45,39 @@ export const registerProductFeatures = (
   const productLines = Object.fromEntries(
     config.productTypes.map((productType) => [productType.product_line, true])
   );
+  const productTiers = Object.fromEntries(
+    config.productTypes.map((productType) => [productType.product_tier, true])
+  );
 
   const configurator: ProductFeaturesConfigurator = {};
   // Cases are always enabled (both for security and AI-SOC)
   configurator.cases = getCasesProductFeaturesConfigurator(enabledProductFeatureKeys);
 
   if (productLines[ProductLine.security]) {
+    // TODO: clarify what happens with siem migrations
+    if (!config.experimentalFeatures.siemMigrationsDisabled) {
+      configurator.siemMigrations =
+        getSiemMigrationsProductFeaturesConfigurator(enabledProductFeatureKeys);
+    }
     configurator.security = getSecurityProductFeaturesConfigurator(
       enabledProductFeatureKeys,
       config.experimentalFeatures
     );
-    // TODO: don't register timeline and notes for security "minimal" tier
-    configurator.timeline = getTimelineProductFeaturesConfigurator(enabledProductFeatureKeys);
-    configurator.notes = getNotesProductFeaturesConfigurator(enabledProductFeatureKeys);
+
+    // timeline and notes are not available for the searchAiLake tier within Security
```

Comment:
Leaving so we do not forget: 
<img width="1898" alt="Screenshot 2025-03-04 at 16 13 41" src="https://github.com/user-attachments/assets/1d8ee584-6190-48ee-9aee-e271459124f9" />
using capabilities do not affect RBAC ( still visible ). What we should do is to not register the features in the first place.